### PR TITLE
Add Reference tools.build.cross_building.can_run

### PR DESCRIPTION
**Pull request by uilianries**
Related to https://github.com/conan-io/conan/pull/11321

Screenshot 2022-05-25 at 14-04-43 conan tools build — conan 1 48 1 documentation

### CODE REVIEW

File: reference/conanfile/tools/build.rst [hunk 1]
```diff
@@ -20,3 +20,22 @@ Parameters:
...
+
+conan.tools.build.can_run()
+----------------------------------
+
+.. code-block:: python
+
+    def can_run(conanfile):
+
+
+Check it we are cross execute any application based on :ref:`global conf <global_conf>`  or ``cross_building()``.
```
Comment (franramirez688) [inline reference/conanfile/tools/build.rst:33]:
```suggestion
Validates whether is possible to run a non-native app on the same architecture.
```

File: reference/conanfile/tools/build.rst [hunk 2]
```diff
@@ -20,3 +20,22 @@ Parameters:
...
+Returns the configuration value for ``tools.build.cross_building:can_run``. In case is not configured,
+it returns ``False`` in the case that we are cross-building, otherwise, ``True``.
```
Comment (franramirez688) [inline reference/conanfile/tools/build.rst:35]: I would rephrase it a little bit:

```
It returns the configuration value for ``tools.build.cross_building:can_run`` if exists, otherwise, it returns ``False`` if we are cross-building, else, ``True``.
```

File: reference/conanfile/tools/build.rst [hunk 3]
```diff
@@ -20,3 +20,22 @@ Parameters:
...
+
+It's an useful feature in case your archtecture can run more then one target. For instance, Mac M1 machines can run both ``armv8`` and ``x86_64``.
```
Comment (franramirez688) [inline reference/conanfile/tools/build.rst:37]:
```suggestion
It's an useful feature for the case your architecture can run more than one target. For instance, Mac M1 machines can run both ``armv8`` and ``x86_64``.
```

**Comment by uilianries** [comment]
Thanks @franramirez688! Much better now!

**Review by franramirez688 — APPROVED** [review]
